code/modules/reagents/chems/chems_medicines.dm

```diff
@@ -359,3 +359,43 @@
 /decl/material/liquid/oxy_meds/affect_blood(var/mob/living/M, var/alien, var/removed, var/datum/reagents/holder)
 	M.add_chemical_effect(CE_OXYGENATED, 1)
 	holder.remove_reagent(/decl/material/gas/carbon_monoxide, 2 * removed)
+
+#define DETOXIFIER_EFFECTIVENESS 6 // 6u of opiates removed per 1u of detoxifier; 5u is enough to remove 30u, i.e. an overdose
+#define DETOXIFIER_DOSE_EFFECTIVENESS 2 // 2u of metabolised opiates removed per 1u of detoxifier; will leave you vulnerable to another OD if you use more
+/decl/material/liquid/detoxifier
```

Comment:
I feel like this behavior should be rolled down into antitoxins, as it's largely a refinement of what that chem does and the names are similar enough to potentially be confusing.